### ISSUE TITLE
Fix `Value.compare`

### ISSUE THREAD

**Issue by filipeom** [formalsec/smtml#210]

**Comment by hra687261** [comment]
Can you give more details on what the issue with it is?

**Comment by filipeom** [comment]
Yes of course. In `Value.compare` the case when two values are not of the same type is undefined and just does: `assert false`. 

https://github.com/formalsec/smtml/blob/30dc99f9257b6a51f5cfbb57e3a12c81ff2b0bf0/src/ast/value.ml#L46-L49

This isn't a massive issue because we don't have a `Expr.compare` yet. But even if we had to have an `Expr.compare` we  we could just use the hash-consing tags to provide a total ordering.
